rcl_yaml_param_parser/src/parse.c

```diff
@@ -801,13 +1083,19 @@ rcutils_ret_t parse_file_events(
               yaml_event_delete(&event);
               return RCUTILS_RET_ERROR;
             }
-            if (0U == params_st->params[node_idx].num_params) {
+            if (0U == params_st->params[node_idx].num_params && map_level != 3U) {
```

Comment:
@bpwilcox meta: since `ros__parameters` and `ros__parameter_descriptors` are siblings, perhaps we should have another flag to signal which one we're processing as opposed to repurposing `map_level`